### PR TITLE
feat: add jerryscript

### PR DESCRIPTION
**Pull request by gocarlos**
Specify library name and version:  **lib/1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/jerryscript/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True
+    }
+    generators = "cmake"
```
Comment (madebr) [inline recipes/jerryscript/all/conanfile.py:22]:
```suggestion
    generators = "cmake"
    short_paths = True
```

**Comment by madebr** [comment]
The `CMAKE_BUILD_TYPE` does not look right in these logs.
Maybe they are hardcoded?

File: recipes/jerryscript/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
```
Comment (uilianries) [inline recipes/jerryscript/all/conanfile.py:59]:
```suggestion
```

Automatically treated by CMake helper

**Comment by uilianries** [comment]
@solvingj I didn't know about your script!

**Comment by solvingj** [comment]
image

**Pull request merged**